File: bindings/java/examples/SignExample.java
import ua.dstucrypto.dstucore.Sign;

/** {@code crypto_sign} (DSTU 4145): generate a signing keypair, sign a message, verify it. */
final class SignExample {
    private SignExample() {
    }

    static void run() throws Exception {
        byte[] signingKey = Sign.keygen();
        byte[] verifyingKey = Sign.verifyingKey(signingKey);

        byte[] message = "a message whose origin and integrity matter".getBytes("UTF-8");
        byte[] signature = Sign.sign(signingKey, message);
        if (!Sign.verify(verifyingKey, message, signature)) {
            throw new IllegalStateException("verification failed");
        }
        System.out.println("signed and verified a " + message.length + "-byte message");

        if (!Sign.verify(verifyingKey, "a different message".getBytes("UTF-8"), signature)) {
            System.out.println("signature over a different message correctly rejected");
        }
    }
}

File: bindings/dotnet/examples/SignExample.cs
using System.Text;
using DstuCore;

namespace DstuCore.Examples;

/// <summary><c>crypto_sign</c> (DSTU 4145): generate a signing keypair, sign a message, verify it.</summary>
internal static class SignExample
{
    public static void Run()
    {
        using var signingKey = SigningKey.Generate();
        using var verifyingKey = signingKey.VerifyingKey();

        var message = Encoding.ASCII.GetBytes("a message whose origin and integrity matter");
        var signature = signingKey.Sign(message);
        if (!verifyingKey.Verify(message, signature))
        {
            throw new InvalidOperationException("verification failed");
        }

        Console.WriteLine($"signed and verified a {message.Length}-byte message");

        if (!verifyingKey.Verify(Encoding.ASCII.GetBytes("a different message"), signature))
        {
            Console.WriteLine("signature over a different message correctly rejected");
        }
    }
}
